src/components/MainView/MainView.jsx

```diff
@@ -3,13 +3,49 @@ import { Container, Col, Row } from 'react-bootstrap';
 import '../../App.scss';
 import CreateDashboard from './CreateDashboard/CreateDashboard';
 import ListOfDashboards from './ListOfDashboards/ListOfDashboards';
+import { db } from '../../fire';
+import getDashArrayFromDb from './getDashArrayFromDb';
 import NavBar from '../Dashboard/Header/Header';
+import WelcomeView from '../Dashboard/Header/WelcomeView';
 
 class MainView extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoggedIn: props.isLoggedIn,
+      dashboards: [],
+    };
+  }
+
+  componentDidMount() {
+    const dashboardsRef = db.database().ref('dashboards');
+    dashboardsRef.on('value', (snapshot) => {
+      const dashboardsSnap = snapshot.val();
+      this.setState({
+        dashboards: getDashArrayFromDb(dashboardsSnap),
+      });
+    });
+  }
+
+  addDashboard = (data) => {
+    this.setState(prevState => ({
+      dashboards: [...prevState.dashboards, data],
+    }));
+  }
+
+  deleteDashboard = (data) => {
```

Comment:
This component uses Redux for state management and this function is already deleted. No need to restore it.